notebooks/DenseClusImplentation.ipynb

```diff
@@ -0,0 +1,1718 @@
ZytO/vxEjlGCmNdCEIJRJ0BcOc7OqlalE5H37/zXg8Drq6g/zzg49TUV/Ftbdupcqw07HrBG+9cTvXfvJtRE62EHz9DRRD0B1P8JuDJwrbFUIwMBhhYDDC4aMto86NqqhsWtTA29avxum0Y2gKh7u7aA4OUFZVxrbtGygt9ZOOJ4j0DVLq8+DzubA7rNicVuwOK6qqYLNZKbVZKRg56s5xR6VD5g/QuKGUJWuuI9wVYvlggv6eMM2v7mfRuqUYbacQg70YTjdUTj3zbLJodhv+dSsJ7j3E4O6D2CtKsQYmNusnW80WMLaKsinFMQGmW7ZqIXSehmwaOk5B9UIYEW8hdJ3onn0Ft5qqKSiqRmTX7kKWjn1BHap9+jFbmsd0g+Si0WlvQyIphpFOFRILNP/URZK1ppZMawt6OISayYt4q51ccBAtUDKl4O2h8gMilZx2qQ/J7KGIC6nCl+dDH/oQXq+Xb3zjG4VpmUyGq6++mj/90z/l4x//eNH12tvbue2223jwwQe5/fbbC9N/9atf8Vd/9Vfs3LmzILqmiq4bDA5O3KkZwNANkgNhwm099B1tIdbZh8Njw18ToGHrchwOUPThgLrB/gQvPXqAzqZ+KgJO6iq8qIpCdyTJW77wITwBM3aio6mDH/7Nt3jbtpW47FbebOrmcHMvd977VirqynFaLCTae+l+8zTZRBpVUVBU0FQVq0VFHeFSyeUM0lmdTFY3C1RmdcpX1HPDH95NLpmm89cvkwlGsFeUYC0rIRuJE2zuglwWbbIXnKZid1nx13gpqw+gqArpeIamN7qJZ3SCySTH2to52dxGKJEkZwhqq0rZuHYJt1x/FYsXmhaDts5BBi1+1l+7sXCjiPUGefbLD+H32lm6dgEiZn4ujtpKKm7ehmK1kE6lyWVzoCioqoqmqaiqSigYpr25g+bTrZw52cKxQyewJHO88+p11OczqTK5HMfO9tLU009fJE46o5vFPg0dh9XK0tIqHBYbKKBaFDKqjm4RqFYVX5mXleuWseHaq0jG47z6/KsEvC4WLayhfkEVVm38m52eNUDoiJwBTjf+BQuIGxZ0oYyKl5oJhGHQ85tXSPcMoNptlGxdh6uhtlBPaNS4kkn6f/MSIpvFf80mnPV109tpNg1dzeb/gOLx462tIZZRCO07SLa3D8ViwbVqBckzLYWHDoCttgbP+rXDDTyngR6LEXrxFVBVSm6/FdU6+8HbFotKSYmbYDA+ulXQZcSVcIww8XGmm8+QaWlG8/txbdo8re2nzjRhdHegaiqGbqCnzZcD1e3BtmgxlooivRKLIIQgvW8X6Dlsq9ejen1UVEw9AFwyO8yISLr22mt573vfy+c+97lR09/+9rezYcMGvvSlLxVd76WXXuJTn/oUTz755Chr08GDB3nf+97HT3/6UzZvnt4XWAiBYZxzaEYOBAjdAM6ZpygTfKEF2axOLp0DQ5jLAkOpX7ousLkdKIaRD2AGgTAbFgKGEOR0gciPR1FAGZk3pipYnXYsNgvCEOjZHEYmO6Ly8blNScdMKSQhicI/Q+dhaIJpAzNjRsBis5jjMP8x5444fj2nk0npIAQ2pxVFHX3M5t8jAqERGLowzw+MLtosjNFjwownsnmcE2ZPhUMRclmzTo4QAj3fKds8ZSp2qxW7RRsjBM3lzTENjWNoNyM/h+FTJUZ/G4bWBXRhYLFoWK0aVqsFLS/gzn/zO2fHYz6x6SGGvmMjKXyG5rGPPOhpvZXmz3dh84V/xsP8Qgpj9L6BC34rFsbQtTrBsSiY9b1mAEUBVVUxDGMqhccvKa6EY4TixymyWfM7lb83KzbblIW80HNmdez8vR7y31BFQZxzrytcm4piZrGNc98Q2ay5TQWwWLBMIyNUMjvMWEySzzfW/O/z+QiHw0XWMBmad+66/rz5c6J1z4eiKGjnWgG06X/xNDtwkRJsrHYrMPniZrOBBbDP8ctMaVlgbgcguWJRrwCXx5VwjHDOcWozULLjAp4j429T9mybr8zqVTLZSrnnLjP0Fiur7EokEolEIpkrZkQk+Xw+IpGxzVSj0WhRC9MQ41mMhrY10boSiUQikUgks8mMiKSlS5fS1NQ0alomk6GtrW3CzLYlS5YAcObMmVHTm5qaUBSlMF8ikUgkEonkYjMjIunGG29k165dBIPBwrRnnnmGTCbDTTfdNO569fX1LFmyhCeffHLU9F//+tesW7du2pltEolEIpFIJBfKjIik97///Xi9Xu677z5eeeUVHn30Uf7+7/+ed7zjHaMsSX/913/N6tWrR637x3/8xzz11FP867/+K6+//jr/8A//wI4dO/jjP/7jmRiaRCKRSCQSybSYkew2n8/HD3/4Qx544AHuv/9+HA4Hd99995iSAIZhoOuj+3vdddddpFIpvvnNb/Ld736XhQsX8q//+q9zVm1bIpFIJBKJBGaoTpJEIpFIJBLJ5caVUShDIpFIJBKJZIpIkSSRSCQSiURSBCmSJBKJRCKRSIogRZJEIpFIJBJJEaRIkkgkEolEIimCFEkSiUQikUgkRZAiSSKRSCQSiaQIM1JMcj6i6waDg/G5HsaEqKpCaambwcE4hnH5lqu6Eo5THuPlgTzGy4dL+TgrKrzTXlcIMS+Peb59HpM9x9KSNIeoqoKiKKiqMtdDmVWuhOOUx3h5II/x8uFKOc5zma/HfKl+HlIkSSQSiUQikRRBiiSJRCKRSCSSIkiRJJFIJBKJRFIEKZIkEolEIpFIiiBFkkQikUgkEkkRLtsSABKJRCKZGwzDQNdzczwGhVRKI5NJo+tzn3IOoGkWVFXaJi4lpEiSSCQSyYwghCASGSSZjM31UADo71cxDGOuhzEKp9ODz1eKolxaqfBXKlIkSSQSiWRGGBJIHk8JNpt9zoWApinzxookhCCTSROLBQHw+8vmeESSySBFkkQikUguGMPQCwLJ4/HN9XAAsFhUcrn5Y0my2ewAxGJBvN6SWXG9Gbo+49
```

Comment:
adding descriptive stats here may help defend why intersection_union_mapper yielded best results - can talk through the clusters and their intuitive meanings